Frameworks/CoreText/DWriteWrapper_CoreText.mm

```diff
@@ -97,97 +108,130 @@ bool _CloneDWriteGlyphRun(_In_ DWRITE_GLYPH_RUN const* src, _Out_ DWRITE_GLYPH_R
     if (settings) {
-        CTTextAlignment alignment = kCTNaturalTextAlignment;
-        if (CTParagraphStyleGetValueForSpecifier(settings, kCTParagraphStyleSpecifierAlignment, sizeof(CTTextAlignment), &alignment)) {
-            DWRITE_TEXT_ALIGNMENT dwriteAlignment = __CoreTextAlignmentToDwrite(alignment);
-            RETURN_NULL_IF_FAILED(textFormat->SetTextAlignment(dwriteAlignment));
-        }
-
-        CTWritingDirection direction;
-        if (CTParagraphStyleGetValueForSpecifier(settings, kCTParagraphStyleSpecifierBaseWritingDirection, sizeof(direction), &direction)) {
-            DWRITE_READING_DIRECTION dwriteDirection = DWRITE_READING_DIRECTION_LEFT_TO_RIGHT;
-            if (direction == kCTWritingDirectionRightToLeft) {
-                dwriteDirection = DWRITE_READING_DIRECTION_RIGHT_TO_LEFT;
-
-                // DWrite alignment is based upon reading direction whereas CoreText alignment is constant
-                // so we have to flip the writing direction
-                if (alignment == kCTRightTextAlignment || alignment == kCTNaturalTextAlignment) {
-                    RETURN_NULL_IF_FAILED(textFormat->SetTextAlignment(DWRITE_TEXT_ALIGNMENT_LEADING));
-                } else if (alignment == kCTLeftTextAlignment) {
-                    RETURN_NULL_IF_FAILED(textFormat->SetTextAlignment(DWRITE_TEXT_ALIGNMENT_TRAILING));
-                }
-            }
-
-            RETURN_NULL_IF_FAILED(textFormat->SetReadingDirection(dwriteDirection));
-        }
-
-        CTLineBreakMode lineBreakMode;
-        if (CTParagraphStyleGetValueForSpecifier(settings,
-                                                 kCTParagraphStyleSpecifierLineBreakMode,
-                                                 sizeof(lineBreakMode),
-                                                 &lineBreakMode)) {
-            DWRITE_WORD_WRAPPING wrapping;
-            switch (lineBreakMode) {
-                // TODO 1121:: DWrite does not support line breaking by truncation, so just use clipping for now
-                case kCTLineBreakByTruncatingHead:
-                case kCTLineBreakByTruncatingTail:
-                case kCTLineBreakByTruncatingMiddle:
-                case kCTLineBreakByClipping:
-                    wrapping = DWRITE_WORD_WRAPPING_NO_WRAP;
-                    break;
-                case kCTLineBreakByCharWrapping:
-                    wrapping = DWRITE_WORD_WRAPPING_CHARACTER;
-                    break;
-                case kCTLineBreakByWordWrapping:
-                default:
-                    wrapping = DWRITE_WORD_WRAPPING_WRAP;
-                    break;
-            }
-
-            THROW_IF_FAILED(textFormat->SetWordWrapping(wrapping));
-        }
+        RETURN_IF_FAILED(__DWriteTextFormatApplyParagraphStyle(textFormat, settings));
     }
 
-    return textFormat;
+    *outTextFormat = textFormat.Detach();
```

Comment:
@aballway `CopyTo` adds an unnecessary +1 on the pointed-to COM object that's -1'd when the method returns immediately after. `*x = y.Detach()` is the idiomatic way to commute a +1 reference to a caller.